front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -246,6 +246,51 @@ const createTrainSchedulePayload = async ({
   };
 };
 
+const handleUpdateTrainSchedule = async ({
+  netzgrafikDto,
+  trainrun,
+  dispatch,
+  infraId,
+  addUpsertedTrainSchedules,
+}: {
+  netzgrafikDto: NetzgrafikDto;
+  trainrun: Trainrun;
+  dispatch: AppDispatch;
+  infraId: number;
+  addUpsertedTrainSchedules: (trainSchedules: TrainScheduleResult[]) => void;
+}) => {
+  const { nodes, labels } = netzgrafikDto;
+  const trainrunSectionsByTrainrunId = getTrainrunSectionsByTrainrunId(netzgrafikDto, trainrun.id);
+  const trainrunIdToUpdate = createdTrainrun.get(trainrun.id) || trainrun.id;
+  const trainSchedule = await dispatch(
+    osrdEditoastApi.endpoints.getTrainScheduleById.initiate({
+      id: trainrunIdToUpdate,
+    })
+  ).unwrap();
+  const startDate = new Date(trainSchedule.start_time);
+  const newTrainSchedule = await dispatch(
+    osrdEditoastApi.endpoints.putTrainScheduleById.initiate({
+      id: trainrunIdToUpdate,
+      trainScheduleForm: {
+        ...trainSchedule,
+        ...(await createTrainSchedulePayload({
+          trainrunSections: trainrunSectionsByTrainrunId,
+          nodes,
+          trainrun,
+          infraId,
+          dispatch,
+          labels,
+          oldStartDate: startDate,
+          trainSchedule,
+        })),
+        // Reset margins because they contain references to path items
+        margins: undefined,
+      },
+    })
+  ).unwrap();
```

Comment:
```suggestion
const trainSchedulePayload = await createTrainSchedulePayload({
    trainrunSections: trainrunSectionsByTrainrunId,
    nodes,
    trainrun,
    infraId,
    dispatch,
    labels,
    oldStartDate: startDate,
    trainSchedule,
  });
  const newTrainSchedule = await dispatch(
    osrdEditoastApi.endpoints.putTrainScheduleById.initiate({
      id: trainrunIdToUpdate,
      trainScheduleForm: {
        ...trainSchedule,
        ...trainSchedulePayload,
        // Reset margins because they contain references to path items
        margins: undefined,
      },
    })
  ).unwrap();
```